File: Complex-Number-Multiplication/CNM.py
class Solution(object):

    def complexNumberMultiply(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        aL = a.split("+", 1)
        bL = b.split("+", 1)
        print(aL, bL)
        ins = int(aL[0]) * int(bL[0]) - int(aL[1][:-1]) * int(bL[1][:-1])
        com = int(aL[0]) * int(bL[1][:-1]) + int(bL[0]) * int(aL[1][:-1])

        if ins != "0":
            return str(ins) + "+" + str(com) + "i"
        else:
            return str(com) + "i"

File: Complex-Number-Multiplication/CNM.rs
fn complex_number_multiply<'a>(x: &'a str, y: &'a str) -> String {
    let mut x_temp: Vec<&str> = x.split('+').collect();
    let mut y_temp: Vec<&str> = y.split('+').collect();

    println!("{:?}", x_temp);
    println!("{:?}", y_temp);
    println!("{:?}", y_temp[1].as_bytes()[1]);

    let ins = x_temp[0].parse::<u32>().unwrap() * y_temp[0].parse::<u32>().unwrap();
    //let com = y_temp
    println!("{:?}", ins);

    return String::from("aa");
}

fn main() {
    complex_number_multiply("2+1i", "3+1i");
}
